hypershift-operator/controllers/nodepool/openstack/openstack.go

```diff
@@ -27,5 +28,29 @@ func MachineTemplateSpec(hcluster *hyperv1.HostedCluster, nodePool *hyperv1.Node
 		return nil, fmt.Errorf("image name is required")
 	}
 
+	port := capiopenstack.PortOpts{}
+	ingressIP, err := getIngressIP(hcluster.Spec.Networking.MachineNetwork[0])
```

Comment:
should this always be required then? regardless if BYON or CAPO managed subnet will be used

Reply:
even if we have the default CIDR specified [here](https://github.com/openshift/hypershift/pull/4423/files#diff-0de67b02a8275f7a8227b3af5101786e70e5c651b852a87f6160f6563a9071d6R85) when no subnets are defined, we would get the machineNetwork empty here, right?

Reply:
The line removal of `machineNetworks = []hyperv1.MachineNetworkEntry{{CIDR: *ipnet.MustParseCIDR(defaultCIDRBlock)}}` was an error, nice catch.

Reply:
fixed, thanks